### PR TITLE
feat(impl): [#528] Improvements Policy Store Input Validation

### PR DESCRIPTION
**Pull request by dsmf**
## Description

- story: https://github.com/eclipse-tractusx/item-relationship-service/issues/528

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [ ] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [ ] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Review by ds-jhartmann — CHANGES_REQUESTED** [review]

**Comment by ds-jhartmann** [comment]
1 PMD violation https://github.com/eclipse-tractusx/item-relationship-service/actions/runs/9035341234/job/24829878885?pr=597#step:5:9503

File: irs-policy-store/src/main/java/org/eclipse/tractusx/irs/policystore/models/UpdatePolicyRequest.java [hunk 1]
```diff
@@ -27,18 +27,23 @@
...
 /**
  * Request object for policy update
  */
 @Schema(description = "Request to update a policy")
+@Builder
 public record UpdatePolicyRequest(
 
         @Schema(description = "Timestamp after which the policy will no longer be accepted in negotiations.") //
         @NotNull //
+        @Future(message = "must be in future") //
```
[the remark this answers is not in the document]
Comment (dsmf) [reply irs-policy-store/src/main/java/org/eclipse/tractusx/irs/policystore/models/UpdatePolicyRequest.java:46]: the message is automatically prefixed with the field name by the framework:
grafik

File: irs-policy-store/src/main/java/org/eclipse/tractusx/irs/policystore/models/CreatePolicyRequest.java [hunk 2]
```diff
@@ -29,19 +29,23 @@
...
  */
 @SuppressWarnings("FileTabCharacter")
 @Schema(description = "Request to add a policy")
+@Builder
 public record CreatePolicyRequest(
 
         @Schema(description = "Timestamp after which the policy will no longer be accepted in negotiations.",
                 example = "2025-12-12T23:59:59.999Z") //
         @NotNull //
+        @Future(message = "must be in future") //
```
[the remark this answers is not in the document]
Comment (dsmf) [reply irs-policy-store/src/main/java/org/eclipse/tractusx/irs/policystore/models/CreatePolicyRequest.java:48]: the message is automatically prefixed with the field name by the framework:
grafik

**Review by ds-jhartmann — APPROVED** [review]
LGTM

**Pull request merged**